libs/vkd3d/utils.c

```diff
@@ -391,13 +391,24 @@ void vkd3d_format_compatibility_list_add_format(struct vkd3d_format_compatibilit
     unsigned int i;
     bool found = false;
 
+    /* Already marked overflow. */
+    if (list->format_count > ARRAY_SIZE(list->vk_formats))
+        return;
+
     for (i = 0; i < list->format_count && !found; i++)
         found = list->vk_formats[i] == vk_format;
 
     if (!found)
     {
-        assert(list->format_count < ARRAY_SIZE(list->vk_formats));
-        list->vk_formats[list->format_count++] = vk_format;
+        if (list->format_count < ARRAY_SIZE(list->vk_formats))
+        {
+            list->vk_formats[list->format_count++] = vk_format;
+        }
+        else
+        {
+            /* Mark overflow. Caller can fall back to plain MUTABLE_FORMAT_BIT without format list. */
+            list->format_count = VKD3D_FORMAT_COMPAT_LIST_ALL_FORMATS;
```

Comment:
Not sure I like this design - might just want to `return false` on overflow and print a warning or something, I feel like that's safer and easier to reason about without having to be aware of the format_count quirk.

Reply:
Would it make more sense to use formatCount == 0 as the sentinel?

Reply:
Updated.